File: Chat-WebServer/AM-Chat-WebServer/Data/DTOs/MessageDTO.cs
﻿using AM_Chat_WebServer.Data.Models;

namespace AM_Chat_WebServer.Data.DTOs;

public class MessageDTO
{
    public long Id { get; set; }
    public long SenderId { get; set; }
    public long ChannelId { get; set; }
    public string Data { get; set; } = string.Empty;
    public MessageType MessageType { get; set; }
    public long Created { get; set; }
    public long? Updated { get; set; } = null;
}

File: app/src/main/java/pwr/barwa/chat/data/dto/MessageDto.kt
package pwr.barwa.chat.data.dto

data class MessageDto(
    val id: Long,
    val senderId: Long,
    val channelId: Long,
    val data: String,
    private val messageType: Int?, // Zmieniamy typ z MessageType na Int z możliwością null
    val created: Long,
    val updated: Long? = null
) {
    // Dodajemy właściwość, która konwertuje Int na MessageType
    val type: MessageType
        get() = when(messageType) {
            null -> MessageType.TEXT // Domyślna wartość gdy null
            1 -> MessageType.TEXT
            2 -> MessageType.IMAGE
            3 -> MessageType.VIDEO
            4 -> MessageType.AUDIO
            else -> MessageType.TEXT // Domyślna dla nieznanych wartości
        }
}

enum class MessageType(val value: Int) {
    TEXT(1),
    IMAGE(2),
    VIDEO(3),
    AUDIO(4)
}

